subprojects/persistent-cache/src/main/java/org/gradle/cache/internal/DefaultPersistentDirectoryCache.java

```diff
@@ -85,14 +85,20 @@ public boolean requiresInitialization(FileLock lock) {
                 return true;
             }
 
-            Properties cachedProperties = GUtil.loadProperties(propertiesFile);
-            for (Map.Entry<?, ?> entry : properties.entrySet()) {
-                String previousValue = cachedProperties.getProperty(entry.getKey().toString());
-                String currentValue = entry.getValue().toString();
-                if (!previousValue.equals(currentValue)) {
-                    LOGGER.debug("Invalidating {} as cache property {} has changed from {} to {}.", DefaultPersistentDirectoryCache.this, entry.getKey(), previousValue, currentValue);
+            if (!properties.isEmpty()) {
```

Comment:
⭕️ Just a stylistic comment: I generally prefer a quick return instead of a long nested if.